lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControlModule.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.security;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.ConfigurationException;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import io.airlift.bootstrap.Bootstrap;
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.HttpClientConfig;
+import io.trino.spi.QueryId;
+import io.trino.spi.connector.CatalogSchemaTableName;
+import io.trino.spi.security.Identity;
+import io.trino.spi.security.SystemAccessControl;
+import io.trino.spi.security.SystemSecurityContext;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestFileBasedSystemAccessControlModule
```

Comment:
What is the purpose of this testing ? Instead of initializing the module we could configure a `LocalQueryRunner` with specific properties.

Reply:
The core part of this test is this line: accessControl.checkCanCreateView(ALICE, aliceView);
We make sure here that rules from file-based-system-catalog.json file have been applied.
Regarding LocalQueryRunner it is not available in this module. It is in trino-main which depends on trino-plugin-toolkit. So this kind of test would have to be created somewhere else trino-main I guess? I like this test as it is quite short and simple. Also because no dependency to LocalQueryRunner can be put close to Module implementation. Not saying the LocalQueryRunner based test is not good idea here but maybe we should have both? WDYT?

Reply:
I'm not sure if we need to test a module...We could do something like `TestFileBasedSystemAccessControl`

Reply:
Actually you are right. TestFileBasedSystemAccessControl covers everything that is tested in TestFileBasedSystemAccessControlModule( It also instantiate module via factory class). Ok, will get rid of this and create two tests for File and Rest based on TestFileBasedSystemAccessControl

Reply:
There are two test classes now -  TestRestBasedSystemAccessControl and TestFileBasedSystemAccessControl. Both extend the same base class